test/models/services.ts

```diff
@@ -425,6 +425,45 @@ describe('services.chooseServiceConfig and services.buildService', function () {
       });
     });
   });
+
+  describe('when the collection match includes variables for variable-based service', function () {
```

Comment:
Add a test for requesting 'all' variables to document whether we return an error or not.

Add a test for specifying a variable that exists, but is not configured in the variable list within services.yml.

Reply:
So, if my [above reply](https://github.com/nasa/harmony/pull/131#discussion_r793050209) makes sense, do we still need a test for `all`?